src/app/clusters/software-diagnostics-server/software-diagnostics-logic.h

```diff
@@ -0,0 +1,101 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/AttributeValueEncoder.h>
+#include <app/data-model-provider/MetadataTypes.h>
+#include <clusters/SoftwareDiagnostics/Enums.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/ReadOnlyBuffer.h>
+#include <platform/DiagnosticDataProvider.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+/// Type-safe implementation for callbacks for the SoftwareDiagnostics server
+class SoftwareDiagnosticsLogic
+{
+public:
+    SoftwareDiagnosticsLogic()          = default;
+    virtual ~SoftwareDiagnosticsLogic() = default;
+
+    CHIP_ERROR GetCurrentHeapFree(uint64_t & out) const { return GetDiagnosticDataProvider().GetCurrentHeapFree(out); }
+    CHIP_ERROR GetCurrentHeapUsed(uint64_t & out) const { return GetDiagnosticDataProvider().GetCurrentHeapUsed(out); }
+    CHIP_ERROR GetCurrentHighWatermark(uint64_t & out) const
```

Comment:
Why are these inlined?

Reply:
Seemed trivial enough and I expected potentially smaller code. Will check code size to validate that assumption.

Reply:
Compare sais non-inline is 72 bytes extra:

![image](https://github.com/user-attachments/assets/3b1a193f-ac27-4de2-8849-45f7a29aae53)

May not be accurate, however it does seem to make sense that adding an extra call chain for these would take up some flash.